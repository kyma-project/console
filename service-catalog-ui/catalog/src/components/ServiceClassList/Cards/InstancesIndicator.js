import React from 'react';
import { isStringValueEqualToTrue } from '../../../commons/helpers';
import { Tooltip } from '@kyma-project/react-components';
import { CardIndicator, CardIndicatorGeneral } from './styled';

const tooltipDescription = {
  provisionOnlyOnce:
    'You can provision this Service Class only once in a given Namespace.',
  provisionOnlyOnceActive:
    'You can provision this Service Class only once in a given Namespace. It is already provisioned in this Namespace.',
  instancesTooltipInfo: 'This Service Class is provisioned',
  instancesTooltipSingle: 'time.',
  instancesTooltipPlural: 'times.',
};

export function InstancesIndicator({ numberOfInstances, labels }) {
  const isProvisionedOnlyOnce =
    labels &&
    labels.provisionOnlyOnce &&
    isStringValueEqualToTrue(labels.provisionOnlyOnce);
  return (
    <CardIndicator>
      {isProvisionedOnlyOnce && (
        <Tooltip
          content={
            numberOfInstances > 0
              ? tooltipDescription.provisionOnlyOnceActive
              : tooltipDescription.provisionOnlyOnce
          }
        >
          <CardIndicatorGeneral
            data-e2e-id="card-indicator"
            provisionOnce
            active={numberOfInstances ? 'true' : 'false'}
          >
            1
          </CardIndicatorGeneral>
        </Tooltip>
      )}
      {!isProvisionedOnlyOnce && numberOfInstances > 0 && (
        <Tooltip
          content={`${
            tooltipDescription.instancesTooltipInfo
          } ${numberOfInstances} ${
            numberOfInstances > 1
              ? tooltipDescription.instancesTooltipPlural
              : tooltipDescription.instancesTooltipSingle
          }`}
        >
          <CardIndicatorGeneral
            data-e2e-id={`instances-provisioned`}
            active={numberOfInstances ? 'true' : 'false'}
          >
            {numberOfInstances}
          </CardIndicatorGeneral>
        </Tooltip>
      )}
    </CardIndicator>
  );
}
