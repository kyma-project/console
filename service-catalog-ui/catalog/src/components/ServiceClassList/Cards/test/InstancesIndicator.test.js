import renderer from 'react-test-renderer';
import React from 'react';
import { InstancesIndicator } from '../InstancesIndicator';

describe('InstancesIndicator', () => {
  it('Renders without instances and provision only once', () => {
    const component = renderer.create(
      <InstancesIndicator
        numberOfInstances={0}
        labels={{ provisionOnlyOnce: 'true', local: 'true' }}
      />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders with instances and provision only once', () => {
    const component = renderer.create(
      <InstancesIndicator
        numberOfInstances={1}
        labels={{ provisionOnlyOnce: 'true', local: 'true' }}
      />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders without instances and no provision only once', () => {
    const component = renderer.create(
      <InstancesIndicator numberOfInstances={0} labels={{ local: 'true' }} />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders with instances and no provision only once', () => {
    const component = renderer.create(
      <InstancesIndicator numberOfInstances={3} labels={{ local: 'true' }} />,
    );
    expect(component).toMatchSnapshot();
  });
});
