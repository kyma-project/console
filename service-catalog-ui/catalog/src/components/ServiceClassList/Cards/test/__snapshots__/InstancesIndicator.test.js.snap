// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstancesIndicator Renders with instances and no provision only once 1`] = `
<div
  class="fd-panel__actions instances-indicator"
>
  <div
    class="sc-jqCOkK fzAxWh"
    type="default"
  >
    <div
      class="instances-indicator--general active"
    >
      3
    </div>
  </div>
</div>
`;

exports[`InstancesIndicator Renders with instances and provision only once 1`] = `
<div
  class="fd-panel__actions instances-indicator"
>
  <div
    class="sc-jqCOkK fzAxWh"
    type="default"
  >
    <div
      class="instances-indicator--general active provision-once"
    >
      1
    </div>
  </div>
</div>
`;

exports[`InstancesIndicator Renders without instances and no provision only once 1`] = `
<div
  class="fd-panel__actions instances-indicator"
/>
`;

exports[`InstancesIndicator Renders without instances and provision only once 1`] = `
<div
  class="fd-panel__actions instances-indicator"
>
  <div
    class="sc-jqCOkK fzAxWh"
    type="default"
  >
    <div
      class="instances-indicator--general provision-once"
    >
      1
    </div>
  </div>
</div>
`;
